api/envoy/api/v2/route/route.proto

```diff
@@ -933,6 +933,10 @@ message RetryPolicy {
   // the base interval. The documentation for :ref:`config_http_filters_router_x-envoy-max-retries`
   // describes Envoy's back-off algorithm.
   RetryBackOff retry_back_off = 8;
+
+  // HTTP headers that trigger a retry if present in the response (the value doesn't matter).
+  // The field is only consulted if 'retriable-headers' retry policy is active.
+  repeated string retriable_headers = 9;
```

Comment:
Can we future proof this by using a repeated HeaderMatcher and related utility functions for this? 

Note this change will require release notes.

Reply:
Ah, didn't know about HeaderMatcher, seems like a great fit for this, thanks Matt!

Reply:
Do you have any preference on how to handle retriable headers provided by the request headers? Some possibilities include:

1) support a comma-separated list of header names and construct key-only HeaderMatcher's for them;
2) disallow specifying them via request headers at all;
3) provide some serialization mechanism for HeaderMatcher.

I like options 1 and 2, as 3 seems too difficult for a user to craft  but don't really have strong feelings about that.